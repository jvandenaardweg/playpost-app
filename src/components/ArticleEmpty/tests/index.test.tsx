import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { ArticleEmptyProcessing, ArticleEmptyFailed } from '../index';

describe('ArticleEmpty', () => {
  let wrapper: RenderAPI;

  describe('ArticleEmptyProcessing', () => {

    describe('rendering default', () => {

      beforeAll(() => {
        wrapper = render(<ArticleEmptyProcessing isLoading={false} url="https://www.google.nl" />);
      });

      it('should render correctly when isLoading is false', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
      });
    });

    describe('rendering loading', () => {

      beforeAll(() => {
        wrapper = render(<ArticleEmptyProcessing isLoading={true} url="https://www.google.nl" />);
      });

      it('should render correctly when isLoading is true', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
      });
    });

  });

  describe('ArticleEmptyFailed', () => {

    describe('rendering default', () => {

      beforeAll(() => {
        wrapper = render(<ArticleEmptyFailed isLoading={false} url="https://www.google.nl" />);
      });

      it('should render correctly', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
      });
    });

    describe('rendering loading', () => {

      beforeAll(() => {
        wrapper = render(<ArticleEmptyFailed isLoading={true} url="https://www.google.nl" />);
      });

      it('should render correctly when isLoading is true', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
      });
    });

  });

});
