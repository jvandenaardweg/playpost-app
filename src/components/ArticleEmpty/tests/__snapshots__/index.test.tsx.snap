// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleEmpty ArticleEmptyFailed rendering default should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F1F1F1",
        "flex": 1,
        "flexDirection": "row",
        "padding": 14,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#E9110B",
      },
    ]
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 34,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 15,
            "fontWeight": "600",
            "lineHeight": 21,
            "marginBottom": 10,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Failed to process an article
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "marginBottom": 10,
            "opacity": 0.9,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Please remove the article from your playlist and try again.
    </Text>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000000",
            "marginBottom": 10,
            "opacity": 0.9,
          },
          Object {
            "color": "#ffffff",
          },
          Object {
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      https://www.google.nl
    </Text>
  </View>
</View>
`;

exports[`ArticleEmpty ArticleEmptyFailed rendering loading should render correctly when isLoading is true 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F1F1F1",
        "flex": 1,
        "flexDirection": "row",
        "padding": 14,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#E9110B",
      },
    ]
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 34,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 15,
            "fontWeight": "600",
            "lineHeight": 21,
            "marginBottom": 10,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Failed to process an article
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "marginBottom": 10,
            "opacity": 0.9,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Please remove the article from your playlist and try again.
    </Text>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000000",
            "marginBottom": 10,
            "opacity": 0.9,
          },
          Object {
            "color": "#ffffff",
          },
          Object {
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      https://www.google.nl
    </Text>
  </View>
</View>
`;

exports[`ArticleEmpty ArticleEmptyNew rendering default should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F1F1",
      "flex": 1,
      "flexDirection": "row",
      "padding": 14,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 34,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 15,
          "fontWeight": "600",
          "lineHeight": 21,
          "marginBottom": 10,
        }
      }
    >
      Not processing article, yet!
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "marginBottom": 10,
          "opacity": 0.9,
        }
      }
    >
      Hold on, it seems busy. We will start processing this article shortly.
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "height": 30,
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "marginBottom": 10,
              "opacity": 0.9,
            },
            Object {
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Check for update
      </Text>
    </View>
  </View>
</View>
`;

exports[`ArticleEmpty ArticleEmptyNew rendering loading should render correctly when isLoading is true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F1F1",
      "flex": 1,
      "flexDirection": "row",
      "padding": 14,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 34,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 15,
          "fontWeight": "600",
          "lineHeight": 21,
          "marginBottom": 10,
        }
      }
    >
      Not processing article, yet!
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "marginBottom": 10,
          "opacity": 0.9,
        }
      }
    >
      Hold on, it seems busy. We will start processing this article shortly.
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "height": 30,
          "justifyContent": "flex-start",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#000000"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</View>
`;

exports[`ArticleEmpty ArticleEmptyProcessing rendering default should render correctly when isLoading is false 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F1F1",
      "flex": 1,
      "flexDirection": "row",
      "padding": 14,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 34,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 15,
          "fontWeight": "600",
          "lineHeight": 21,
          "marginBottom": 10,
        }
      }
    >
      Processing article...
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "marginBottom": 10,
          "opacity": 0.9,
        }
      }
    >
      The article from "
      www.google.nl
      " is still being processed.
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "height": 30,
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "marginBottom": 10,
              "opacity": 0.9,
            },
            Object {
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Check for update
      </Text>
    </View>
  </View>
</View>
`;

exports[`ArticleEmpty ArticleEmptyProcessing rendering loading should render correctly when isLoading is true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F1F1",
      "flex": 1,
      "flexDirection": "row",
      "padding": 14,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 34,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 15,
          "fontWeight": "600",
          "lineHeight": 21,
          "marginBottom": 10,
        }
      }
    >
      Processing article...
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "marginBottom": 10,
          "opacity": 0.9,
        }
      }
    >
      The article from "
      www.google.nl
      " is still being processed.
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "height": 30,
          "justifyContent": "flex-start",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#000000"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</View>
`;
